test/exe/terminate_handler_test.cc

```diff
@@ -0,0 +1,13 @@
+#include "exe/terminate_handler.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+TEST(TerminateHandler, HandlerInstalledTest) {
+  TerminateHandler handler;
+  EXPECT_DEATH([]() -> void { std::terminate(); }(),
+               ".*Envoy::TerminateHandler::logOnTerminate().*|backtrace\\.h");
+}
```

Comment:
I was going to ask if you could write a test that actually throws, but I tried that and gmock catches it first and helpfully indicates the test failed due to unhandled exception :) Oh well...

Reply:
I tried to do this originally too. :-(